src/api/Nhs.Appointments.Core/BookingWriteService.cs

```diff
@@ -42,12 +45,14 @@ Task<BookingRecalculationsStatistics> RecalculateAppointmentStatuses(string site
 public class BookingWriteService(
     IBookingsDocumentStore bookingDocumentStore,
     IBookingQueryService bookingQueryService,
-    IReferenceNumberProvider referenceNumberProvider,
+    IReferenceNumberProvider referenceNumberProviderV1,
+    IProvider referenceNumberProviderV2,
```

Comment:
Very minor point but is there a reason we're naming the new interface so generically? It has methods like `bool IsValidBookingReference(string bookingReference);` so doesn't seem to be a generic thing. Would it be clearer to name it `IReferenceNumberProvider`?

If the rename is just because of a naming clash, you could either use import aliases or just have both providers implement the same interface. V1 would just need to throw NotImplementedExceptions for the new methods